src/components/Dashboard/widgets/Wallet/TxConfirmation.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { ReactElement, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import styled from '@emotion/styled';
+
+import link from 'assets/link.svg';
+import { dispatch } from 'store/rematch';
+import {
+  selectAccountAddress,
+  selectCurrentTxError,
+  selectCurrentTxId,
+  selectCurrentTxStatus,
+} from 'store/selectors';
+import { formatFiat, limitLength } from 'util/helpers';
+import links from 'util/links';
+import { Colors } from 'vars/defines';
+
+import { Button } from 'components/_General/buttons';
+import CopyToClipboard from 'components/_General/CopyToClipboard';
+import ErrorMessage from 'components/_General/ErrorMessage';
+import { GrayLabel } from '../common';
+import TxConfirmationRow from './TxConfirmationRow';
+
+const TxConfirmationRoot = styled.div`
+  height: 340px;
+`;
+const Row = styled.div`
+  display: flex;
+  flex-direction: row;
+  margin-bottom: 8px;
+`;
+
+const Column = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  margin-bottom: 8px;
+`;
+type TxConfirmationProps = {
+  currency?: string;
+  recepient: string;
+  amount: string;
+  usdValue?: string;
+};
+
+const TxConfirmation = ({
+  currency,
+  recepient,
+  amount,
+  usdValue,
+}: TxConfirmationProps): ReactElement => {
+  const txid = useSelector(selectCurrentTxId);
+  const address = useSelector(selectAccountAddress);
+  const txStatus = useSelector(selectCurrentTxStatus);
+  const txError = useSelector(selectCurrentTxError);
+
+  const [currentTxId, setCurrentTxId] = useState(null);
+
+  const usdValueTemp = formatFiat(Number(amount) * Number(usdValue));
+
+  if (txStatus === 1) {
```

Comment:
This kind of thing should probably be wrapped in a `useEffect` so that it is not run on every re-render.